src/azure-cli-core/azure/cli/core/commands/parameters.py

```diff
@@ -107,12 +107,33 @@ def enum_choice_list(data):
     def _type(value):
         return next((x for x in choices if x.lower() == value.lower()), value) if value else value
     params = {
-        'choices': CaseInsenstiveList(choices),
+        'choices': CaseInsensitiveList(choices),
         'type': _type
     }
     return params
 
 
+def three_state_flag(none_default=True, use_enabled_disabled=False):
```

Comment:
Don't think `none_default` and `use_enabled_disabled` are clear names.
Suggest a more descriptive name or method comments describing the args.